core/commands/keystore.go

```diff
@@ -688,6 +691,128 @@ func keyOutputListEncoders() cmds.EncoderFunc {
 	})
 }
 
+type KeySignOutput struct {
+	Key       KeyOutput
+	Signature string
+}
+
+var keySignCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
```

Comment:
nit: let's set expectations  that this api may change:
```suggestion
	Status: cmds.Experimental,
	Helptext: cmds.HelpText{
```